### PR TITLE
ssh-agent: stax port

### PR DESCRIPTION
**Pull request by cbouvet-ledger**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -55,24 +58,33 @@ DEFINES   += HAVE_U2F HAVE_IO_U2F
...
+DEFINES       += IO_SEPROXYHAL_BUFFER_SIZE_B=300
 DEFINES       += HAVE_BLE BLE_COMMAND_TIMEOUT_MS=2000
 DEFINES       += HAVE_BLE_APDU # basic ledger apdu transport over BLE
+SDK_SOURCE_PATH  += lib_blewbxx lib_blewbxx_impl
 endif
 
-ifeq ($(TARGET_NAME),TARGET_NANOS)
-DEFINES       += IO_SEPROXYHAL_BUFFER_SIZE_B=128
+ifneq (,$(filter $(TARGET_NAME),TARGET_NANOS TARGET_NANOS2))
+DEFINES   += IO_SEPROXYHAL_BUFFER_SIZE_B=128
```
Comment (xchapron-ledger) [inline Makefile:70]: Nanos2 should have 300B
Something like:
```
ifeq ($(TARGET_NAME),TARGET_STAX)
DEFINES   += NBGL_QRCODE
else
DEFINES   += HAVE_BAGL HAVE_UX_FLOW
ifneq ($(TARGET_NAME),TARGET_NANOS)
DEFINES   += HAVE_GLO096
DEFINES   += BAGL_WIDTH=128 BAGL_HEIGHT=64
DEFINES   += HAVE_BAGL_ELLIPSIS # long label truncation feature
DEFINES   += HAVE_BAGL_FONT_OPEN_SANS_REGULAR_11PX
DEFINES   += HAVE_BAGL_FONT_OPEN_SANS_EXTRABOLD_11PX
DEFINES   += HAVE_BAGL_FONT_OPEN_SANS_LIGHT_16PX
endif
endif

ifeq ($(TARGET_NAME),$(filter $(TARGET_NAME),TARGET_NANOX TARGET_STAX))
DEFINES   += HAVE_BLE BLE_COMMAND_TIMEOUT_MS=2000 HAVE_BLE_APDU
endif

ifeq ($(TARGET_NAME),TARGET_NANOS)
DEFINES       += IO_SEPROXYHAL_BUFFER_SIZE_B=128
else
DEFINES       += IO_SEPROXYHAL_BUFFER_SIZE_B=300
endif

ifneq ($(TARGET_NAME),TARGET_STAX)
SDK_SOURCE_PATH += lib_ux
endif

ifeq ($(TARGET_NAME),$(filter $(TARGET_NAME),TARGET_NANOX TARGET_STAX))
SDK_SOURCE_PATH += lib_blewbxx lib_blewbxx_impl
endif
```

But do you really need QRCODE for NBGL?

File: Makefile [hunk 2]
```diff
@@ -121,10 +133,11 @@ include $(BOLOS_SDK)/Makefile.glyphs
 ### computed variables
 APP_SOURCE_PATH  += src
 SDK_SOURCE_PATH  += lib_stusb lib_stusb_impl lib_u2f
-SDK_SOURCE_PATH  += lib_ux
 
-ifeq ($(TARGET_NAME),TARGET_NANOX)
-SDK_SOURCE_PATH  += lib_blewbxx lib_blewbxx_impl
+ifeq  ($(TARGET_NAME),TARGET_STAX)
+SDK_SOURCE_PATH  += lib_ux_stax
```
Comment (xchapron-ledger) [inline Makefile:138]: not needed, see proposition above.

File: Makefile [hunk 3]
```diff
@@ -55,24 +58,33 @@ DEFINES   += HAVE_U2F HAVE_IO_U2F
...
 
-ifeq ($(TARGET_NAME),TARGET_NANOS)
-DEFINES       += IO_SEPROXYHAL_BUFFER_SIZE_B=128
+ifneq (,$(filter $(TARGET_NAME),TARGET_NANOS TARGET_NANOS2))
+DEFINES   += IO_SEPROXYHAL_BUFFER_SIZE_B=128
+endif
+
+ifeq  ($(TARGET_NAME),TARGET_STAX)
+DEFINES   += HAVE_NBGL
+DEFINES   += NBGL_QRCODE
```
Comment (xchapron-ledger) [inline Makefile:75]: needed?

File: Makefile [hunk 4]
```diff
@@ -55,24 +58,33 @@ DEFINES   += HAVE_U2F HAVE_IO_U2F
...
 endif
 
-ifeq ($(TARGET_NAME),TARGET_NANOS)
-DEFINES       += IO_SEPROXYHAL_BUFFER_SIZE_B=128
+ifneq (,$(filter $(TARGET_NAME),TARGET_NANOS TARGET_NANOS2))
+DEFINES   += IO_SEPROXYHAL_BUFFER_SIZE_B=128
+endif
+
+ifeq  ($(TARGET_NAME),TARGET_STAX)
+DEFINES   += HAVE_NBGL
```
Comment (xchapron-ledger) [inline Makefile:74]: not needed.

File: src/ssh_ux_nbgl.c [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+#include "nbgl_use_case.h"
+#include "main.h"
+
+void app_quit() {
+  io_seproxyhal_touch_exit(NULL);
+}
+
+void ui_idle(void) {
+  nbgl_useCaseHome(APPNAME,
+                    NULL,
```
Comment (xchapron-ledger) [inline src/ssh_ux_nbgl.c:15]: No icon?

File: src/ssh_ux_nbgl.c [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+                    APPNAME,
```
Comment (xchapron-ledger) [inline src/ssh_ux_nbgl.c:16]: "Application is ready" to be euqivalent with current Nano Ux?

File: src/ssh_ux_nbgl.c [hunk 7]
```diff
@@ -0,0 +1,95 @@
+#ifdef HAVE_NBGL
+
+#include "os_helpers.h"
+#include "ux.h"
+#include "callbacks.h"
+#include "nbgl_use_case.h"
+#include "main.h"
+
+void app_quit() {
+  io_seproxyhal_touch_exit(NULL);
```
Comment (xchapron-ledger) [inline src/ssh_ux_nbgl.c:10]:
```suggestion
  os_sched_exit(-1);
```

File: Makefile (hunk 1 above)
Comment (cbouvet-ledger) [reply Makefile:70]: i dont

File: .gitignore [hunk 8]
```diff
@@ -1,7 +1,6 @@
+build
 bin
 debug
-dep
-obj
 src/u2f_crypto_data.h
 src/glyphs.h
 src/glyphs.c
```
Comment (xchapron-ledger) [inline .gitignore:6]: Can be dropped too.
```suggestion
```

File: .gitignore [hunk 9]
```diff
@@ -1,7 +1,6 @@
+build
 bin
 debug
-dep
-obj
 src/u2f_crypto_data.h
```
Comment (xchapron-ledger) [inline .gitignore:4]: Looks like a bad copy-paste from app-u2f?
```suggestion
```

File: Makefile [hunk 10]
```diff
@@ -46,32 +48,49 @@ all: default
...
-DEFINES       += HAVE_BAGL_FONT_OPEN_SANS_LIGHT_16PX
+endif
+
+ifneq ($(TARGET_NAME),TARGET_STAX)
+SDK_SOURCE_PATH += lib_ux
+endif
+
+ifneq (,$(filter $(TARGET_NAME),TARGET_NANOX TARGET_STAX))
+DEFINES       += HAVE_BLE BLE_COMMAND_TIMEOUT_MS=2000
+DEFINES       += HAVE_BLE_APDU # basic ledger apdu transport over BLE
```
Comment (xchapron-ledger) [inline Makefile:92]: Duplicate with above.
```suggestion
```

File: src/ssh_ux_nbgl.c [hunk 11]
```diff
@@ -6,12 +6,31 @@
...
 
+static const char *const infoTypes[]    = {"Version", "Developer", "Copyright"};
+static const char *const infoContents[] = {APPVERSION, "Ledger", "(c) 2023 Ledger"};
+
+static bool navigation_cb(uint8_t page, nbgl_pageContent_t *content) {
+    UNUSED(page);
+    content->type                   = INFOS_LIST;
+    content->infosList.nbInfos      = 3;
+    content->infosList.infoTypes    = (const char **) infoTypes;
+    content->infosList.infoContents = (const char **) infoContents;
```
Comment (xchapron-ledger) [inline src/ssh_ux_nbgl.c:23]:
```suggestion
    content->infosList.infoTypes    = infoTypes;
    content->infosList.infoContents = infoContents;
```

File: src/ssh_ux_nbgl.c [hunk 12]
```diff
@@ -6,12 +6,31 @@
...
+    return true;
+}
+
+void app_settings() {
+    nbgl_useCaseSettings("SSH/PGP Agent", 0, 1, false, ui_idle, navigation_cb, NULL);
```
Comment (xchapron-ledger) [inline src/ssh_ux_nbgl.c:28]:
```suggestion
    nbgl_useCaseSettings(APPNAME, 0, 1, false, ui_idle, navigation_cb, NULL);
```

**Review by xchapron-ledger — DISMISSED** [review]

**Review by xchapron-ledger — APPROVED** [review]

**Pull request merged**